scanpipe/pipelines/deploy_to_develop.py

```diff
@@ -152,6 +128,15 @@ def flag_whitespace_files(self):
         """Flag whitespace files with size less than or equal to 100 byte as ignored."""
         d2d.flag_whitespace_files(project=self.project)
 
+    def load_ecosystem_config(self):
+        """Load ecosystem specific configurations for d2d steps for selected options."""
+        d2d.load_ecosystem_config(pipeline=self, options=self.selected_groups)
+
+    @optional_step("Ruby")
+    def load_ecosystem_config_ruby(self):
+        """Load Ruby specific configurations for d2d steps."""
+        pass
```

Comment:
Is this a leftover or are we planning to duplicate those methods?

Reply:
Ruby is the only option which does not require any code to match source/binary as we already have sha1 matching working there correctly. But the loading of ecosystem configurations and showing/selecting these options in the UI happens on the basis of having optional steps, so I'm having a dummy step here to enable loading the config and showing this as an option everywhere that we can select.

Reply:
Ruby should not be special. I get that the mapping is already done with sha1matching, but I think you should not name this "load_ecosystem_config_ruby" as a "tag" step for ruby: this is confusing. I suggest instead to rename this to "map_ruby": it may not do anything just now, but it will likely do things later as explained in:
https://github.com/aboutcode-org/scancode.io/issues/1436#issuecomment-3018881766

therefore the "pass" there will be temporary.